tests/samplers_tests/test_samplers.py

```diff
@@ -229,6 +242,142 @@ def sample() -> float:
     np.testing.assert_almost_equal(round_points, points)
 
 
+@parametrize_relative_sampler
+@pytest.mark.parametrize(
+    "x_distribution",
+    [
+        UniformDistribution(-1.0, 1.0),
+        LogUniformDistribution(1e-7, 1.0),
+        DiscreteUniformDistribution(-10, 10, 0.5),
+        IntUniformDistribution(1, 10),
+        IntLogUniformDistribution(1, 100),
+    ],
+)
+@pytest.mark.parametrize(
+    "y_distribution",
+    [
+        UniformDistribution(-1.0, 1.0),
+        LogUniformDistribution(1e-7, 1.0),
+        DiscreteUniformDistribution(-10, 10, 0.5),
+        IntUniformDistribution(1, 10),
+        IntLogUniformDistribution(1, 100),
+    ],
+)
+def test_sample_relative_numerical(
+    relative_sampler_class: Callable[[], BaseSampler],
+    x_distribution: BaseDistribution,
+    y_distribution: BaseDistribution,
+) -> None:
+
+    search_space: Dict[str, BaseDistribution] = OrderedDict(x=x_distribution, y=y_distribution)
+    study = optuna.study.create_study(sampler=relative_sampler_class())
+    trial = study.ask(search_space)
+    study.tell(trial, sum(trial.params.values()))
+
+    def sample() -> List[float]:
```

Comment:
A nit but we're also parameterizing over int distributions.
```suggestion
    def sample() -> List[Union[int, float]]:
```